### ISSUE TITLE
[webview_flutter] Autofill doesn't work

### ISSUE THREAD

**Issue by MihaMarkic** [flutter/flutter#92165]
## Steps to Reproduce

1. Run sample provided in this [comment](https://github.com/flutter/flutter/issues/67526#issuecomment-896834729).

**Expected results:** 

If Autofill is configured properly, it should suggest e-mail. It does with the same page both in natively hosted WebView and in browser (tested on Android 11 emulator and 10 on a device)

**Actual results:** 

No autofill suggestions.

<details>
<summary>Code sample</summary>

```html
<input autocomplete="email" id="email" name="email" type="email">
```

(copied from steps to reproduce hyperlink)
```dart
import 'dart:async';
import 'dart:convert';
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() => runApp(MaterialApp(home: WebViewExample()));

class WebViewExample extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    if (Platform.isAndroid) WebView.platform = SurfaceAndroidWebView();

    return Scaffold(
      appBar: AppBar(),
      body: WebView(
        initialUrl:
            'data:text/html;base64,PCFET0NUWVBFIGh0bWw+CjxodG1sPgo8aGVhZD48dGl0bGU+VGVzdCBhdXRvIGNvbXBsZXRlPC90aXRsZT48L2hlYWQ+Cjxib2R5PgplbWFpbDoKPGlucHV0IGF1dG9jb21wbGV0ZT0iZW1haWwiIGlkPSJlbWFpbCIgbmFtZT0iZW1haWwiIHR5cGU9ImVtYWlsIj4KPC9ib2R5Pgo8L2h0bWw+Cg==',
        javascriptMode: JavascriptMode.unrestricted,
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
```

```
Analyzing android...
No issues found! (ran in 0.7s)
```

```
[✓] Flutter (Channel stable, 2.5.2, on Microsoft Windows [Version 10.0.19043.1288], locale sl-SI)
    • Flutter version 2.5.2 at C:\Users\MYNAME\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (3 weeks ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\MYNAME\AppData\Local\Android\Sdk
    • Platform android-31, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\MYNAME\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio
    • Android Studio at C:\Program Files\Android\Android Studio\
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = C:\Program Files\Android\Android Studio\
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.61.1)
    • VS Code at C:\Users\MYNAME\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.27.0
```

</details>

**Comment by danagbemava-nc** [comment]
I tested the sample in a native app and a flutter app and the behaviour is the same in both apps. See videos below.

`flutter sample`

https://user-images.githubusercontent.com/88313112/138228209-3e47385f-d333-4a64-bfd6-62703b0b9032.mp4 

`native sample`

https://user-images.githubusercontent.com/88313112/138228245-f2a61571-80a5-47d1-8cdd-cfaf0a50f1c8.mp4


Can you show the behaviour you're experiencing?

**Comment by dbarrera-fsl** [comment]
Are there any news on this? We've been trying to get autofill working on our app and haven't been able to.

**Comment by mattsolle-sf** [comment]
We're experiencing this issue in our production application (MGM Resorts International). Would love an update on this item.

**Comment by SuperDoris** [comment]
> > On Xamarin which uses webview directly, autofill is triggered on all fields correctly
> 
> `Webview_flutter` is a wrapper around the native webview, so ideally, it should be able to do whatever the native webviews can do. I guess a few features slip through the cracks here and there.
> 
> I tested with your link and with the native android webview, the emails popup, but it doesn't popup in the flutter webview.
> 
> Labelling for further investigation.
> 
> screenshots
> flutter doctor -v

I encountered the same problem. I used 'adb shell uiautomator dump' to capture the view contents of both the native WebView and the Flutter WebView. There are differences between them. The native WebView is under the node of ViewGroup, while the Flutter WebView is under the node of View. Both cases can trigger the Android Autofill Service, but when the service is triggered by the native WebView, it can traverse the WebView and its HTML info, while the service is triggered by the Flutter WebView, it cannot traverse the WebView nodes and its HTML info.

Therefore, I think the problem is caused by the way the Flutter engine renders the WebView. Here are my screenshots and the results of my code.
Here are some relevant screenshots:
![flutter_webview](https://user-images.githubusercontent.com/107848195/221393515-23eac86a-d533-4fac-adf9-3203f482395c.png)
![native_android](https://user-images.githubusercontent.com/107848195/221393521-e64faa0c-b5de-44ca-a5ef-73c698561248.png)

![screenshot_flutter_webview_treenode](https://user-
![screenshot_native_viewnode](https://user-images.githubusercontent.com/107848195/221393550-4fac6097-b77c-4182-ba39-3dcbc9d8f43a.png)
images.githubusercontent.com/107848195/221393547-d49bbc5e-fabd-4cfb-9322-a1a518a3c6a7.png)

**Comment by blast3rz** [comment]
We're having the same issue here. Autofill doesn't work in Android WebView. It worths re-evaluating the priority of this issue since Autofill not working can be a major UX impacting issue.
CC: @stuartmorgan

**Comment by bparrishMines** [comment]
I did some investigating and I think the issued may be caused by the PlatformView implementation. For some reason, requests for autofill doesn't seem to trigger automatically for the `WebView`. 

I was able to trigger one manually after the page was loaded with:

```java
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
  final AutofillManager manager = getContext().getSystemService(AutofillManager.class);
  manager.requestAutofill(webView);
}
```

As a temporary solution, we could add a method to `AndroidWebViewController` that calls the code above. For a long term solution, we would need to investigate why it doesn't trigger automatically like a non-Flutter Android app.
